### PR TITLE
Expose Customization to Dark Reader CSS

### PR DESCRIPTION
**Pull request by Gusted**
Follow-up for #2763

### CODE REVIEW

**Comment by alexanderby** [comment]
Can you explain what does it make?

**Comment by Gusted** [comment]
&{selectionBackground} in the devtools will be replaced by the current background selection that the user chooses. As some like jupyter playground use code mirror that use a class to simulate the selection

**Comment by alexanderby** [comment]
An interesting idea. Maybe let's simplify it. This mechanism with `${}` is weird, people don't understand it. I'm not rushing you to implement it, just to think. Remember we were discussing CSS variables for static themes? What if we declare some variables, like `--darkreader-neutral-bg, --darkreader-neutral-text, --darkreader-selection-bg` etc? Then people can use it in their configs like `.selector { background: var(--darkreader-selection-bg) !important; }`.

**Comment by Gusted** [comment]
Good idea actually with the variables

**Comment by alexanderby** [comment]
Maybe let's start with 4 variables then.

You can declare
```html
<style class="darkreader darkreader--variables">
    :root {
      --darkreader-neutral-bg: ...;
      ...
    }
</style>

File: src/inject/dynamic-theme/index.ts [hunk 1]
```diff
@@ -91,6 +91,20 @@ function createStaticStyleOverrides() {
...
+    variableStyle.textContent = [
+        `:root {`,
+        `   --darkreader-neutral-bg: ${filter.mode === 0 ? lightSchemeBackgroundColor : darkSchemeBackgroundColor};`,
+        `   --darkreader-neutral-text: ${filter.mode === 0 ? lightSchemeTextColor : darkSchemeTextColor};`,
+        `   --darkreader-selection-bg: ${selectionColors.backgroundColorSelection};`,
+        `   --darkreader-selection-fg: ${selectionColors.foregroundColorSelection};`,
+        `}`
+    ].join('\n');
+    document.head.insertBefore(variableStyle, inlineStyle.nextSibling);
+    setupStylePositionWatcher(variableStyle, 'variable');
```
Comment (alexanderby) [inline src/inject/dynamic-theme/index.ts:107]: Let's use plural `variables`

File: src/inject/dynamic-theme/index.ts [hunk 2]
```diff
@@ -91,6 +91,20 @@ function createStaticStyleOverrides() {
     overrideStyle.textContent = fixes && fixes.css ? replaceCSSTemplates(fixes.css) : '';
     document.head.appendChild(overrideStyle);
     setupStylePositionWatcher(overrideStyle, 'override');
+
+    const variableStyle = createOrUpdateStyle('darkreader--variable');
+    const selectionColors = getSelectionColor(filter);
+    const {darkSchemeBackgroundColor, darkSchemeTextColor, lightSchemeBackgroundColor, lightSchemeTextColor} = filter;
+    variableStyle.textContent = [
+        `:root {`,
+        `   --darkreader-neutral-bg: ${filter.mode === 0 ? lightSchemeBackgroundColor : darkSchemeBackgroundColor};`,
```
Comment (alexanderby) [inline src/inject/dynamic-theme/index.ts:100]: How do you think, if it is meant to be used by users, maybe we should use colors like `--darkreader-neutral-background` and `--darkreader-neutral-text`? Maybe `bg` tells something, but `fg` is less common.

File: src/inject/dynamic-theme/index.ts [hunk 3]
```diff
@@ -91,6 +91,20 @@ function createStaticStyleOverrides() {
     overrideStyle.textContent = fixes && fixes.css ? replaceCSSTemplates(fixes.css) : '';
     document.head.appendChild(overrideStyle);
     setupStylePositionWatcher(overrideStyle, 'override');
+
+    const variableStyle = createOrUpdateStyle('darkreader--variable');
```
Comment (alexanderby) [inline src/inject/dynamic-theme/index.ts:95]: Let's use plural `darkreader--variables`

**Pull request merged**